lib/beaker/ca_cert_helper.rb

```diff
@@ -0,0 +1,137 @@
+require 'beaker'
+require 'beaker/dsl/helpers/host_helpers'
+
+module Beaker::CaCertHelper
+
+  def install_ca_certs
+    install_ca_certs_on(hosts)
+  end
+
+
+  ##
+  # Install certs on Windows and OSX Hosts
+  #
+  # @param Array<Host>
+  #
+  ##
+  def install_ca_certs_on(hosts)
+    Array[hosts].each do |host|
+      get_cert_hash.each do |cert_name, ca|
+        create_cert_on_host(host, cert_name, ca)
+        if host['platform'] =~ /windows/i
+          add_windows_cert host, cert_name
+        elsif host['platform'] =~ /osx-10/i
+          add_osx_cert host, cert_name
+        end
+      end
+    end
+  end
+
+  ##
+  # create the remote file and ensure proper rights
+  #
+  def create_cert_on_host(host, cert_name, ca)
+    create_remote_file(host, cert_name, ca)
+    on host, "chmod 644 #{cert_name}"
+  end
+
+  ##
+  # Install cert, assumes cygwin currently
+  ##
+  def add_windows_cert(host, ca_filename)
+    on host, "cmd /c certutil -v -addstore Root `cygpath -w #{ca_filename}`"
+  end
+
+  ##
+  # Add cert to OSX, assumes >= 10.9
+  def add_osx_cert(host, ca_filename)
+    on host, "sudo security add-trusted-cert -d -r trustRoot -k \"/Library/Keychains/System.keychain\" #{ca_filename}"
```

Comment:
Work is underway to include the necessary cacerts for OSX (PUP-4822) so I don't think we want to do this step, or perhaps only do it for puppet versions that are known to not work correctly. Otherwise, this setup step could mask regressions, e.g. we change forgeapi to use UserTrustNetwork, but the OSX package doesn't trust it.